tests/CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
 enable_testing()
-include_directories(../src)
-add_library(testdll SHARED testdll.c)
-set_target_properties(testdll PROPERTIES PREFIX "")
-add_library(testdll2 SHARED testdll2.c)
-set_target_properties(testdll2 PROPERTIES PREFIX "")
-target_link_libraries(testdll2 dl)
-add_library(testdll3 SHARED testdll3.c)
-set_target_properties(testdll3 PROPERTIES PREFIX "")
-add_executable(t_dlfcn test.c)
-target_link_libraries(t_dlfcn dl)
-add_test (NAME t_dlfcn COMMAND t_dlfcn)
+if(WIN32)
+    include_directories(../src)
```

Comment:
Is this `include_directories` really needed? Is not purpose of `target_include_directories` in src to automatically add required include directories for targets?

Reply:
At the dl cmake target there is used  `target_include_directories(dl INTERFACE $<INSTALL_INTERFACE:include>)`, which sets the include path for usage by an external client package, which links to the imported target dlfcn-win32::dl (see https://cmake.org/cmake/help/latest/command/target_include_directories.html for details), but there is no option to set an include path only for clients in the local package, which requires to use that statement.

Reply:
Unless I am missing something, I think we can avoid the `include_directories` by changing
~~~
target_include_directories(dl INTERFACE $<INSTALL_INTERFACE:include>)
~~~
to
~~~
target_include_directories(dl INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                                                                 $<INSTALL_INTERFACE:include>)
~~~

Reply:
Unfortunately no, because paths passed to the INTERFACE property are exported to the installed file dlfcn-win32-targets.cmake and export a path within the build directory from dlfcn-win32 to clients, which will already be deleted at the time of use.

```
set_target_properties(dlfcn-win32::dl PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "/home/xxx/src/dlfcn-win32/src"
)
```

I think this is one of the shortcomings of the CMake design, namely that there is no property for specifying the interface at `target_include_directory', which is only limited to the current CMake project.

PRIVATE = current target
INTERFACE = all others without the current target
PUBLIC = PRIVATE + INTERFACE

There should be a new property, e.g. 

PROJECT_INTERFACE = all clients in the current project